### DISCUSSION TITLE
About the loss function of the model

### DISCUSSION THREAD

**Discussion by Foxyrep** [tvhahn/ml-tool-wear#1] [Q&A] [answered]
Hello, at first thank you for posting this work, and it helps me a lot on my comprehension about TCN and AE! This is what I'm trying to learn these days!
Well there is a question I'm confused about. It's about the loss function in the model compiling.
In the .ipydb file, In[17], Line 173, well since we have set the latent loss, why don't we use the latent loss in the model compiling?  
Hope you could help me on this question, thx!!

**Comment by tvhahn** [comment, marked as answer]
Thanks for the compliment! I'm glad it helped.

We use the `latent_loss` (line 129 in [17]) by using the Keras `add_loss` [method](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#add_loss) (line 138 in [17]). There is a good explanation of why Keras has this "add_loss" method in this stackoverflow post: https://stackoverflow.com/a/52683522/9214620.

Aurélien Geron has an easier to understand implementation of a VAE in his Jupyter notebook. It's worth looking at. His explanation in the book is good too. https://github.com/ageron/handson-ml2/blob/master/17_autoencoders_and_gans.ipynb

Let me know if you have any other questions. Cheers!

**Reply by Foxyrep** [reply]
Grateful for your so quick reply! I'll go through the references you advised later, thx a lot!
